core/authentication/fake.go

```diff
@@ -29,26 +33,50 @@ func NewFake() *FakeAuth {
 	return &FakeAuth{}
 }
 
-// NewContext adds authentication details to a new background context.
-func (a *FakeAuth) NewContext(userID string) context.Context {
-	md := make(map[string][]string)
-	md["userid"] = []string{userID}
-	return metadata.NewOutgoingContext(context.TODO(), md)
+func GetFakeCredential(userID string) credentials.PerRPCCredentials{
```

Comment:
For code clarity, would you mind moving this function below `FakeCredential`? 
We like to keep structs and their associated functions close together. Thanks!